content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-node-definition-reference.md

```diff
@@ -0,0 +1,382 @@
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Key__ | Required | The attribute name to be set | `Key="AZ::Edit::Attributes::GenericValueList"` |
+| __Value__ | Required | The value to set on the attribute | `Value="&amp;BaseTimer::GetTimeUnitList"` |
+
+### Examples
+
+```xml
+<Property Name="m_timeUnits" Type="int" DefaultValue="0" Serialize="true">
+    <PropertyData Name="Units" Description="Units to represent the time in." UIHandler="AZ::Edit::UIHandlers::ComboBox">
+        <EditAttribute Key="AZ::Edit::Attributes::GenericValueList" Value="&amp;BaseTimer::GetTimeUnitList"/>
+        <EditAttribute Key="AZ::Edit::Attributes::PostChangeNotify" Value="&amp;BaseTimer::OnTimeUnitsChanged"/>
+    </PropertyData>
+</Property>
+```
+
+----
+
+
+----
+
+### <a href="#parameter"></a>Parameter
+
+__Parameter__ is used by the [Input](#input) and [Output](#output) elements, they represent data slots on the node.
+ 
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Name__ | | | | 
+| __Type__ | Required | This is the C++ type of the property. It must be a type exposed to the O3DE serialization context. If the type contains special characters (e.g. `AZStd::vector<int>`) you must use the HTML Entity Codes (e.g. `AZStd::vector&lt;int&gt;`) | `Type="float"` |
+| __Description__ | Optional | Displayed as a tooltip when hovering over the slot | `Description="The radius of the circle"`|
+| __DefaultValue__ | Optional | Allows you to provide the default value for the property | `DefaultValue="1000.0"`|
+| __DisplayGroup__ | Optional | Allows slots to be grouped visually | `DisplayGroup="Tests"` |
+----
+
+### Return
+
+__Return__ is used by the [Input](#input) element, represents and output data slots on the node. You can think of it as the return value in a function (i.e. `int Foo() { return 42; }`)
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Name__ | Required | The name will be used as the data output slot's name | `Name="Return Value"` | 
+| __Type__ | Required | This is the C++ type of the property. It must be a type exposed to the O3DE serialization context. If the type contains special characters (e.g. `AZStd::vector<int>`) you must use the HTML Entity Codes (e.g. `AZStd::vector&lt;int&gt;`) | `Type="float"` |
+| __Description__ | Optional | Displayed as a tooltip when hovering over the slot | `Description="The radius of the circle"`|
+| __DefaultValue__ | Optional | Allows you to provide the default value for the property | `DefaultValue="1000.0"`|
+| __DisplayGroup__ | Optional | Allows slots to be grouped visually | `DisplayGroup="Tests"` |
+----
+
+### Examples
```

Comment:
Either add a usage example for Return, or refer the reader to the usage example for Branch.

Reply:
As a reader I wouldn't know I need to go look for it there when I'm reading about this specific topic. I don't think it's a problem if the example is relevant in both areas, I'd prefer to keep each area as self-contained as possible. And when not possible or too redundant I was relying on anchors to help navigate readers to the relevant sections.

Reply:
It's fine to keep examples for Branch and Return separate. This section is missing an example though. Just has the "### Examples" header, then the "### Branch" header in the next line.